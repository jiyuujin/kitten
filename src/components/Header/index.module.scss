.title {
    font-weight: 700;
    font-size: 48px;
    text-transform: uppercase;
    line-height: 1;
    text-shadow: 0 2px 5px rgba(0, 0, 0, 0.3);
    margin: 32px 0;
    max-width: 100%;

    .name {
        font-size: 1.6rem;
    }

    .locale {
        font-size: 0.8rem;
    }
}
