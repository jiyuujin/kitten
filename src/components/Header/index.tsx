import React, { FC } from 'react'
import * as SC from './index.module.scss'

type ProfileHeader = {
    title: string
    author: string
    locale: string
}

export const Header: FC<ProfileHeader> = ({ title, author, locale }) => {
    return (
        <div className={SC.title}>
            {title}
            <div className={SC.name}>
                {author}
            </div>
            <div className={SC.locale}>
                {locale}
            </div>
        </div>
    )
}
