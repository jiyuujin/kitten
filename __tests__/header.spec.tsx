import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../src/components/Header'

const siteMetadata = {
    title: 'Web Developer',
    titleTemplate: '%s | Portfolio%s | ポートフォリオ',
    // eslint-disable-next-line
    description: `こちらは Web Developer \'Yuma Kitamura\' ポートフォリオです。`,
    biography:
        'ネット上では 自由人、jiyuujin と名乗って生息しています。誕生日は1988年11月04日。大阪府豊中市生まれ。性別は男性。現在は関西圏（京都や大阪）を中心に活動しています。',
    position: 'Front-End Engineer,Web Developer,HR Adviser',
    workExperience:
        '自分のキャリアのスタートはiOSネイティブアプリのエンジニアとして。これまでのキャリアの中でiOSネイティブアプリエンジニアとして業務を通じ基礎的なコーディングを取得、Webを扱うサーバサイドで足りていないと言った環境も相まってjQueryを始めとしたCSS設計、フレームワークの使用など徐々にフロントエンドエンジニアという領域にも踏み込んでいきました。',
    author: 'Yuma Kitamura',
    locale: 'Osaka, Kyoto in Japan',
    lonlat: '35.003238° N, 135.750389° E',
    siteUrl: 'https://yuukit.me/',
    image: 'src/images/bakeneko2.png',
    blogName: 'Web猫ブログ',
    blogUrl: 'https://webneko.dev/',
    twitterUsername: '@jiyuujin',
    twitterUrl: 'https://twitter.com/jiyuujinlab',
}

jest.mock('react-intl', () => {
    const reactIntl = jest.requireActual('react-intl')
    const intl = reactIntl.createIntl({
        locale: 'en',
    })
    return {
        ...reactIntl,
        useIntl: () => intl,
    }
})

describe('Header', () => {
    it('render correctly', () => {
        const tree = renderer.create(<Header data={siteMetadata} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
