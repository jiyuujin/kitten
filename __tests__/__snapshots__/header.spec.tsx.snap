// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<div>
  <div
    className="x0 x1 x2 "
  >
    <div
      style={
        Object {
          "height": "auto",
          "transition": "width 150ms linear",
          "width": "auto",
        }
      }
    >
      <div
        className="x0 x3 x4 x5"
        style={
          Object {
            "opacity": 1,
            "position": "relative",
            "transform": "translateY(0px)",
          }
        }
      >
        <div>
          Web Developer ðŸ”¨
        </div>
      </div>
    </div>
  </div>
  <div>
    Yuma Kitamura
  </div>
  <div>
    Osaka, Kyoto in Japan
  </div>
</div>
`;
